src/main/java/com/aws/greengrass/easysetup/DeviceProvisioningHelper.java

```diff
@@ -454,7 +461,8 @@ private Optional<String> getPolicyArn(String policyName, Region awsRegion) {
     }
 
     private String getAccountId() {
-        return stsClient.getCallerIdentity(GetCallerIdentityRequest.builder().build()).account();
+        return stsWrapper.execute(client ->
+                client.getCallerIdentity(GetCallerIdentityRequest.builder().build()).account());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

`software.amazon.awssdk.services.sts.StsClient.getCallerIdentity` API can also throw the following exception types: `SdkClientException`, `StsException`, `SdkException`, `AwsServiceException`, `UnsupportedOperationException`. We recommend handling these uncaught exceptions as well.